### PR TITLE
Make last breadcrumb unclickable

### LINKED ISSUE #735: Don't render last item of breadcrumbs as link
#### Summary
It should not be possible to click on the last item of the breadcrumbs, as it represents the current page.

#### Why do we need this?
Cosmetic UX fix.

#### What is already there? What do you see now?
Breadcrumb component.

#### What is missing? What do you want to see?
The last item rendering as plain `<span />`

#### Environment
Console in browser.

#### How do you propose to implement this?
Self-explanatory

#### Can you do this yourself and submit a Pull Request?
Yes.

### PR DESCRIPTION
**Pull request by pgalic96**
#### Summary

Closes #735 

#### Changes

- Add new `lastBreadcrumb` prop to breadcrumb component
- If `lastBreadcrumb = true` render span instead of `Link` component
- Add custom styling class for the new span
- Fix typo in storybook

### CODE REVIEW

File: pkg/webui/components/breadcrumbs/breadcrumbs.js [hunk 1]
```diff
@@ -20,13 +20,21 @@ import PropTypes from '../../lib/prop-types'
...
 const Breadcrumbs = ({ className, breadcrumbs }) => (
-
+  
   <nav className={classnames(className, style.breadcrumbs)}>
     {breadcrumbs.map(function (component, index) {
-      return (
-        React.cloneElement(component, { key: index })
-      )
-    })}
+      if (index == breadcrumbs.length - 1) {
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumbs.js]:
```suggestion
      if (index === breadcrumbs.length - 1) {
```

File: pkg/webui/components/breadcrumbs/breadcrumbs.js [hunk 2]
```diff
@@ -20,13 +20,21 @@ import PropTypes from '../../lib/prop-types'
...
+        return (
+          React.cloneElement(component, { key: index, lastBreadcrumb: true})
+        )
+      }
+      else {
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumbs.js]: you dont need `else` here

File: pkg/webui/components/breadcrumbs/breadcrumb/index.js [hunk 3]
```diff
@@ -48,6 +57,8 @@ Breadcrumb.propTypes = {
   icon: PropTypes.string,
   /** The content of the breadcrumb */
   content: PropTypes.message.isRequired,
+  /** The flag for rendering last breadcrumb as plain text */
+  lastBreadcrumb: PropTypes.bool,
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumb/index.js]:
```suggestion
  isLast: PropTypes.bool,
```

File: pkg/webui/components/breadcrumbs/breadcrumb/index.js [hunk 4]
```diff
@@ -27,18 +27,27 @@ const Breadcrumb = function ({
...
+    return (
+      <Link
+        to={path}
+        className={classnames(className, style.link)}
+      >
+        {icon && <Icon className={style.icon} icon={icon} />}
+        {isRawText ? <span>{content}</span> : <Message content={content} /> }
+      </Link>
+    )
+  } else {
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumb/index.js]: you dont need `else` here

File: pkg/webui/components/breadcrumbs/breadcrumb/index.js [hunk 5]
```diff
@@ -27,18 +27,27 @@ const Breadcrumb = function ({
...
-    </Link>
-  )
+  if (!lastBreadcrumb) {
+    return (
+      <Link
+        to={path}
+        className={classnames(className, style.link)}
+      >
+        {icon && <Icon className={style.icon} icon={icon} />}
+        {isRawText ? <span>{content}</span> : <Message content={content} /> }
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumb/index.js]: avoid repetition here. try something like this:
```
  let Component
  let componentProps
  if (!lastBreadcrumb) {
    Component = Link
    componentProps = { className: classnames(className, style.link), to: path }
  } else {
    Component = 'span'
    componentProps = { className: classnames(className, style.last) }
  }

  return (
    <Component {...componentProps}>
      {icon && <Icon className={style.icon} icon={icon} />}
      {isRawText ? <span>{content}</span> : <Message content={content} /> }
    </Component>
  )
```

**Comment by pgalic96** [comment]
@kschiffer really no clue why is that happening, if I fix the height it works but that's probably not the best idea

**Comment by kschiffer** [comment]
Have a look at `display: flex` and `align-items: center`. I could imagine the span is not rendering as `display: flex`, which causes the issue.

File: pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl [hunk 6]
```diff
@@ -31,6 +31,12 @@
     text-decoration: underline
     color: $tc-deep-gray
 
+.last
+  one-liner()
+  text-decoration: none
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]: you dont need this

File: pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl [hunk 7]
```diff
@@ -31,6 +31,12 @@
...
+  color: $tc-deep-gray
+  nudge('up')
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]: move this to `L36`

File: pkg/webui/components/breadcrumbs/story.js [hunk 8]
```diff
@@ -19,7 +19,7 @@ import { withInfo } from '@storybook/addon-info'
 import Breadcrumb from './breadcrumb'
 import Breadcrumbs from './breadcrumbs'
 
-storiesOf('Breacrumbs', module)
+storiesOf('Breadcrumbs', module)
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/story.js]: 🚀🚀🚀

File: pkg/webui/components/breadcrumbs/breadcrumbs.js [hunk 9]
```diff
@@ -23,10 +23,16 @@ const Breadcrumbs = ({ className, breadcrumbs }) => (
 
   <nav className={classnames(className, style.breadcrumbs)}>
     {breadcrumbs.map(function (component, index) {
+      if (index === breadcrumbs.length - 1) {
+        return (
+          React.cloneElement(component, { key: index, isLast: true })
```
Comment (bafonins) [inline pkg/webui/components/breadcrumbs/breadcrumbs.js]:
```suggestion
          React.cloneElement(component, { key: index, isLast: index === breadcrumbs.length - 1 })
```

File: pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl [hunk 10]
```diff
@@ -31,6 +31,11 @@
     text-decoration: underline
     color: $tc-deep-gray
 
+.last
+  one-liner()
+  nudge('up')
```
Comment (kschiffer) [inline pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]:
```suggestion
```
Without this, it is positioned correctly.

Comment (pgalic96) [reply pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]: @kschiffer I added `nudge('up')` to fix the issue with alignment when there is no icons. if I remove it the last breadcrumb is not aligned again.

Comment (kschiffer) [reply pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]: I see. I was able to fix the issue by applying a `min-height` to the breadcrumb `<a>` and `<span> // for the last one` and using `display: flex` for them. No more nudging needed then.

Comment (pgalic96) [reply pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]: okay great will do that

Comment (pgalic96) [reply pkg/webui/components/breadcrumbs/breadcrumb/breadcrumb.styl]: I think adding `min-height` alone fixes the issue without `display: flex`

**Pull request merged**